src/GoTrueApi.ts

```diff
@@ -364,6 +365,43 @@ export default class GoTrueApi {
     }
   }
 
+  /**
+   * Send User supplied Email / Mobile OTP to be verified
+   * @param email The user's email address
+   * @param phone The user's phone number WITH international prefix
+   * @param token token that user was sent to their mobile phone
+   * @param type verification type that the otp is generated for
+   * @param redirectTo A URL or mobile address to send the user to after they are confirmed.
+   */
+  async verifyOTP(
+    { email, phone, token, type }: VerifyOTPParams,
+    options: {
+      redirectTo?: string
+    } = {}
+  ): Promise<{ data: Session | User | null; error: ApiError | null }> {
+    try {
+      const headers = { ...this.headers }
+      const data = (email && type) 
+        ? await post(
+          this.fetch,
+          `${this.url}/verify`,
+          { email, token, type: type, redirect_to: options.redirectTo },
+          { headers }
+        )
+        : await post(
+          this.fetch,
+          `${this.url}/verify`,
+          { phone, token, type: 'sms', redirect_to: options.redirectTo },
+          { headers }
+        )
```

Comment:
@thorwebdev i still need to check for `email` and `type` because they will be passed into the object as `{ email: undefined, phone: ..., token: ..., type: undefined }` if i don't